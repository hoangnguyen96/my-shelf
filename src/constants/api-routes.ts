export const API_ROUTES = {
  USER: "/user",
  BOOKS: "/books",
};
