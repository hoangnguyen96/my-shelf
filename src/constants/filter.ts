export enum TYPE_SEARCH {
  TITLE = "title",
  AUTHOR = "author",
}
