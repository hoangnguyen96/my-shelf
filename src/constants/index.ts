export * from "./input";
export * from "./navbar";
export * from "./routes";
export * from "./api-routes";
export * from "./validate";
export * from "./message";
export * from "./filter";
export * from "./time";
