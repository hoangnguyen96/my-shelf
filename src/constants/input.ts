export const PASSWORD = {
  HIDE: "Hide password",
  SHOW: "Show password",
};
