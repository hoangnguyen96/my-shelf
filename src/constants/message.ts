export const MESSAGES = {
  RESPONSE_ERROR: "The server is not responding. Please try again!",
  GET_ERROR: "The server does not respond. Retrieving data failed!",
  POST_ERROR: "The server does not respond. Please try again!",
  DELETE_ERROR: "Delete failed. Please try again!",
  EDIT_ERROR: "Edit failed. Please try again!",
  SUBMIT_ERROR: "Submit failed. Please try again!",
  UPDATE_ERROR: "Update failed. Please try again!",
  CREATE_ERROR: "Create data failed. Please try again!",
  ADD_USER_FAILED: "Failed to add user!",
  REGISTER_SUCCESS: "You have successfully created an account.",
  CREATE_FAILED: "Account creation failed!",

  // Response Api
  LOGIN_SUCCESS: "You have successfully logged in.",
  LOGIN_FAILED: "Login failed!",
  NETWORK_ERROR: "Network response was not ok!",
  EMAIL_EXISTS: "Email exists!",
  EMAIL_PASSWORD_INCORRECT: "Incorrect email or password!",
  EMAIL_PASSWORD_INVALID: "Email or password invalid.Please help to try again!",
  UPLOAD_FAILED: "Failed to upload image!",
  ERROR_UPLOAD: "Error uploading image!",

  // Errors
  LOGIN_NOTFOUND: "User not found or password missing",
  INVALID_CREDENTIALS: "Invalid credentials",

  // Validation
  FIELD_REQUIRED: "This field is required.",

  // Dynamic error messages
  FORMAT: (ariaLabel: string) => `${ariaLabel} does not follow format.`,

  // ConfirmPassword
  CONFIRM_PASSWORD: "Passwords do not match!",
};
