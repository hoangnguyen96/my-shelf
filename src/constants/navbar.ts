export const NAVBAR_STEP = {
  HOME: "Home",
  SEARCH: "Search",
  MY_SHELF: "My Shelf",
  CONTRIBUTE: "Contribute",
};
