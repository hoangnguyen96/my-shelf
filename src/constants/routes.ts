export const ROUTES = {
  LANDING_PAGE: "/",
  DASHBOARD: "/dashboard",
  LOGIN: "/login",
  REGISTER: "/register",
  HOME: "/home",
  SEARCH: "/search",
  PREVIEW: "/preview",
  MY_BOOK_SHELF: "/my-book-shelf",
  MY_BOOK_SHELF_FAVORITES: "/my-book-shelf/favorites",
  CONTRIBUTE: "/contribute",
  CONTRIBUTE_COMPLETE: "/contribute/complete",
  PROFILE: "/profile",
  CONTRIBUTE_LIST: "/contribute-list",
};
