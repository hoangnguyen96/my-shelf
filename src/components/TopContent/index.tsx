"use client";

import { Flex } from "@chakra-ui/react";
import { usePathname } from "next/navigation";
import { ROUTES } from "@app/constants";
import { SearchBar } from "../common";
import MenuProfile from "../MenuProfile";
import { memo } from "react";
import { Session } from "next-auth";

const TopContent = ({ session }: { session?: Session }) => {
  const pathName = usePathname();

  const isNotSearch =
    pathName?.includes(ROUTES.CONTRIBUTE) ||
    pathName?.includes(ROUTES.CONTRIBUTE_COMPLETE) ||
    pathName?.includes(ROUTES.CONTRIBUTE_LIST) ||
    pathName?.includes(ROUTES.PREVIEW) ||
    pathName?.includes(ROUTES.PROFILE);

  return (
    <Flex
      alignItems="center"
      justifyContent={!isNotSearch ? "space-between" : "flex-end"}
      p="32px 44px"
      gap="10px"
    >
      {!isNotSearch && <SearchBar />}
      <MenuProfile session={session} />
    </Flex>
  );
};

export default memo(TopContent);
