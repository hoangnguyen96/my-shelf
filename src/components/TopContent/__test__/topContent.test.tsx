import { render } from "@testing-library/react";
import TopContent from "..";
import { useSession } from "next-auth/react";
import { usePathname } from "next/navigation";
import { ROUTES } from "@app/constants";

jest.mock("next-auth/react", () => ({
  useSession: jest.fn(),
}));

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  usePathname: jest.fn(),
}));

jest.mock("@app/actions/auth", () => ({
  logout: jest.fn(),
}));

describe("TopContent component", () => {
  (useSession as jest.Mock).mockReturnValue({
    data: {
      user: {
        isAdmin: true,
        email: "admin@gmail.com",
        id: "3733403",
        name: "admin",
        image: "https://i.ibb.co/RHMqQGr/man-1.png",
      },
      expires: "2024-12-31T23:59:59.999Z",
    },
    status: "authenticated",
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Should render correctly snapshot", () => {
    const { asFragment } = render(<TopContent />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render path correctly with pathname", () => {
    (usePathname as jest.Mock).mockReturnValue(ROUTES.CONTRIBUTE);
    (usePathname as jest.Mock).mockReturnValue(ROUTES.CONTRIBUTE_COMPLETE);
    (usePathname as jest.Mock).mockReturnValue(ROUTES.CONTRIBUTE_LIST);
    (usePathname as jest.Mock).mockReturnValue(ROUTES.PREVIEW);
    (usePathname as jest.Mock).mockReturnValue(ROUTES.PROFILE);

    const { container } = render(<TopContent />);

    const flexContainer = container.querySelector("div");
    expect(flexContainer).toHaveStyle("justify-content: flex-end");
  });
});
