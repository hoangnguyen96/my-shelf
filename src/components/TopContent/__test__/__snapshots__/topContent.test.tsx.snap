// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopContent component Should render correctly snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1s5jq5f"
  >
    <div
      class="css-19jt6ar"
    >
      <div
        class="chakra-select__wrapper css-18a3ys"
      >
        <select
          aria-label="select-label"
          class="chakra-select css-oea4np"
        >
          <option
            value=""
          >
            Select
          </option>
          <option
            class="css-0"
            value="title"
          >
            Title
          </option>
          <option
            class="css-0"
            value="author"
          >
            Author
          </option>
        </select>
        <div
          class="chakra-select__icon-wrapper css-162mkon"
        >
          <svg
            aria-hidden="true"
            class="chakra-select__icon"
            focusable="false"
            role="presentation"
            style="width: 1em; height: 1em; color: currentColor;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="chakra-input__group css-bx0blc"
        data-group="true"
      >
        <input
          class="chakra-input css-1ufd936"
          placeholder="Search..."
          value=""
        />
        <button
          aria-label="Search"
          class="chakra-button css-xia019"
          data-testid="submit-search"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-1k92xky"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <button
      aria-controls="menu-list-:r1:"
      aria-expanded="false"
      aria-haspopup="menu"
      class="chakra-button chakra-menu__menu-button css-115zmqy"
      id="menu-button-:r1:"
      type="button"
    >
      <span
        class="css-xl71ch"
      >
        <div
          class="css-yvf3fq"
        >
          <div
            class="css-bgqooo"
          >
            <img
              alt="Avatar user"
              data-nimg="fill"
              decoding="async"
              fetchpriority="high"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <p
            class="chakra-text css-1rr4qq7"
          >
            admin
          </p>
        </div>
      </span>
      <span
        class="chakra-button__icon css-1hzyiq5"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-1bho5id"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <div
      class="css-0"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-1r8dsm8"
        id="menu-list-:r1:"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8);"
        tabindex="-1"
      >
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="0"
          id="menu-list-:r1:-menuitem-:r2:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Profile
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="1"
          id="menu-list-:r1:-menuitem-:r3:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Favorites
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="2"
          id="menu-list-:r1:-menuitem-:r4:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Logout
          </p>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
